pkg/test/k8s/namespace.go

```diff
@@ -0,0 +1,73 @@
+package k8s
+
+import (
+	"context"
+	testwait "github.com/codeready-toolchain/toolchain-operator/test/wait"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+)
+
+type NamespaceAssertion struct {
+	namespace      *v1.Namespace
+	client         client.Reader
+	namespacedName types.NamespacedName
+	t              *testing.T
+}
+
+func (a *NamespaceAssertion) loadNamespaceAssertion() error {
+	if a.namespace != nil {
+		return nil
+	}
+	ns := &v1.Namespace{}
+	err := a.client.Get(context.TODO(), a.namespacedName, ns)
+	a.namespace = ns
+	return err
+}
+
+func AssertThatNamespace(t *testing.T, name string, client client.Reader) *NamespaceAssertion {
+	return &NamespaceAssertion{
+		client:         client,
+		namespacedName: types.NamespacedName{Name: name},
+		t:              t,
+	}
+}
+
+func (a *NamespaceAssertion) DoesNotExist() *NamespaceAssertion {
+	err := wait.Poll(testwait.RetryInterval, testwait.Timeout, func() (done bool, err error) {
```

Comment:
So, you want to reuse this in both unit and e2e tests? But in unit tests we don't need to wait (sine we use fake clients). I wonder if we can tune the RetryInterval & Timeout (or add a special flag to avoid polling at all?) for the unit tests so we don't wait (== faster tests).

Reply:
@alexeykazakov That's really good point, it'll be even worse if unit tests is going to fail. you'll have to wait for 1 minute for every tests, which can be avoided>

Reply:
Addressed in 9457b01